blobstore/src/main/java/org/jclouds/blobstore/config/LocalBlobStore.java

```diff
@@ -276,21 +276,11 @@ public StorageMetadata apply(String key) {
          if (options.getMarker() != null) {
             final String finalMarker = options.getMarker();
             String delimiter = storageStrategy.getSeparator();
-            Optional<StorageMetadata> lastMarkerMetadata;
-            if (finalMarker.endsWith(delimiter)) {
-               lastMarkerMetadata = tryFind(contents, new Predicate<StorageMetadata>() {
```

Comment:
I suspect this curious logic is a leftover from the deprecated directory handling.